projects/packages/global-styles-webfonts-introspector/src/class-global-styles-webfonts-introspector.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Global styles webfonts introspector
+ *
+ * @package automattic/global-styles-webfonts-introspector
+ * @since 0.1.0
+ */
+
+namespace Automattic\Jetpack\Fonts;
+
+/**
+ * Global styles webfonts introspector.
+ */
+class Global_Styles_Webfonts_Introspector {
+	/**
+	 * Extract the font family slug from a settings object.
+	 *
+	 * @param object $setting The setting object.
+	 *
+	 * @return string|void
+	 */
+	private static function extract_font_slug_from_setting( $setting ) {
+		if ( isset( $setting['typography'] ) && isset( $setting['typography']['fontFamily'] ) ) {
+			$font_family = $setting['typography']['fontFamily'];
+
+			// Full string: var(--wp--preset--font-family--slug).
+			// We do not care about the origin of the font, only its slug.
+			preg_match( '/font-family--(?P<slug>.+)\)$/', $font_family, $matches );
+
+			if ( isset( $matches['slug'] ) ) {
+				return $matches['slug'];
+			}
+
+			// Full string: var:preset|font-family|slug
+			// We do not care about the origin of the font, only its slug.
+			preg_match( '/font-family\|(?P<slug>.+)$/', $font_family, $matches );
+
+			if ( isset( $matches['slug'] ) ) {
+				return $matches['slug'];
+			}
+
+			return $font_family;
+		}
+	}
+
+	/**
+	 * Introspect global styles for webfonts.
+	 */
+	public static function introspect_global_styles_webfonts() {
+		if ( ! function_exists( 'gutenberg_get_global_styles' ) ) {
+			return;
+		}
+
+		$global_styles = gutenberg_get_global_styles();
+
+		$found_webfonts = array();
+
+		// Look for fonts in block presets...
```

Comment:
`Non-blocking`: How would we feel about providing a little more context for those that might be unfamiliar with global styles?

```
// Global styles can specify typography for individual blocks,
// so look for fonts in block presets...


// Global styles can specify typography for elements like headings, paragraphs, etc.,
// so look for fonts in HTML element presets...
```

Reply:
That's more explicit, for sure! I'm curious if this is necessary since it kind of goes without saying that global styles can specify the typography since we're looking for them in the code?

Reply:
> I'm curious if this is necessary since it kind of goes without saying that global styles can specify the typography since we're looking for them in the code?

Sure -- for those examples, I see what you're saying. My intention is to ultimately provide more clarity for others by adding some context about business use cases, especially since this code will live in Jetpack for the time being where other folks might not be as familiar with global styles.

Does that make sense? Maybe we can think about different, more meaningful wording.

Reply:
Gotcha. Sure, we can think of something.